chia/util/config.py

```diff
@@ -48,10 +48,24 @@ def config_path_for_filename(root_path: Path, filename: Union[str, Path]) -> Pat
     return root_path / "config" / filename
 
 
-def get_config_lock(root_path: Path, filename: Union[str, Path]) -> BaseFileLock:
+@contextlib.contextmanager
+def lock_config(root_path: Path, filename: Union[str, Path]) -> Iterator[None]:
+    # TODO: This is presently used in some tests to lock the saving of the
+    #       configuration file without having loaded it right there.  This usage
+    #       should probably be removed and this function made private.
     config_path = config_path_for_filename(root_path, filename)
     lock_path: Path = config_path.with_name(config_path.name + ".lock")
-    return FileLock(lock_path)
+    with FileLock(lock_path):
+        yield
```

Comment:
By yielding at this point, I'm assuming that an uncaught exception will cause the lock to be released. That definitely sounds like an important improvement.

Reply:
Well, what we returned was used as a context manager so that was covered anyways.  The effect of this is to keep the `FileLock` instance 'private' instead of giving it out to the caller.  This probably doesn't matter, especially in combination with the comment above suggesting this should be a private function, but...  I dunno.  I'm trying to remember why I made this particular change and I don't recall a strong reason.  I'm guessing it was related to some intermediate point in the refactor.  I think it's slightly better, but can also undo it if preferred.